packages/app/src/cli/models/extensions/specification.ts

```diff
@@ -118,19 +123,78 @@ export function createConfigExtensionSpecification<TConfiguration extends BaseCo
   identifier: string
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   schema: zod.ZodObject<any>
-  appModuleFeatures: (config?: TConfiguration) => ExtensionFeature[]
+  appModuleFeatures?: (config?: TConfiguration) => ExtensionFeature[]
+  transformConfig?: TransformationConfig
 }): ExtensionSpecification<TConfiguration> {
+  const appModuleFeatures = spec.appModuleFeatures ?? (() => [])
   return createExtensionSpecification({
     identifier: spec.identifier,
     // This casting is required because `name` and `type` are mandatory for the existing extension spec configurations,
     // however, app config extensions config content is parsed from the `shopify.app.toml`
     schema: spec.schema as unknown as ZodSchemaType<TConfiguration>,
-    appModuleFeatures: spec.appModuleFeatures,
-    transform: defaultAppConfigTransform,
-    reverseTransform: (content) => defaultAppConfigReverseTransform(spec.schema, content),
+    appModuleFeatures: appModuleFeatures().includes('app_config')
+      ? appModuleFeatures
+      : () => appModuleFeatures().concat('app_config'),
+    transform: resolveAppConfigTransform(spec.transformConfig),
+    reverseTransform: resolveReverseAppConfigTransform(spec.schema, spec.transformConfig),
   })
 }
 
+function resolveAppConfigTransform(transformConfig?: TransformationConfig) {
+  if (!transformConfig) return (content: {[key: string]: unknown}) => defaultAppConfigTransform(content)
+
+  return (content: {[key: string]: unknown}) => appConfigTransform(content, transformConfig)
+}
+
+function resolveReverseAppConfigTransform<T>(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  schema: zod.ZodType<T, any, any>,
+  transformConfig?: TransformationConfig,
+) {
+  if (!transformConfig) return (content: {[key: string]: unknown}) => defaultAppConfigReverseTransform(schema, content)
+
+  return (content: {[key: string]: unknown}) => appConfigTransform(content, transformConfig, true)
+}
+
+/**
+ * Given an object:
+ * ```json
+ * { source: { fieldSourceA: 'valueA' } }
+ * ```
+ *  and a tranform config content like this:
+ * ```json
+ * { 'target.fieldTargetA': 'source.fieldSourceA'}
+ * ```
+ * the method returns the following object:
+ * ```json
+ * { source: { fieldSourceA: 'valueA' } }
+ * ```
+ * The tranfomat can be applied in both ways depending on the reverse parameter
```

Comment:
Is that example right? 

```suggestion
 * { source: { fieldTargetA: 'valueA' } }
 * ```
 * The transformation can be applied in both ways depending on the reverse parameter
```